plugins/builtin/target/aws-asg/plugin/aws.go

```diff
@@ -116,15 +116,36 @@ func (t *TargetPlugin) scaleIn(ctx context.Context, asg *types.AutoScalingGroup,
 	// would like on all log lines.
 	log := t.logger.With("action", "scale_in", "asg_name", *asg.AutoScalingGroupName)
 
+	// Check if policy overrides the plugin configuration for
+	// scale_in_protection.
+	scaleInProtection := t.scaleInProtectionEnabled
+	if str, ok := config[configKeyScaleInProtection]; ok {
+		b, err := strconv.ParseBool(str)
+		if err != nil {
+			return fmt.Errorf("failed to parse %s value from policy: %v", configKeyScaleInProtection, err)
+		}
+		scaleInProtection = b
+	}
+	log = log.With("scale_in_protection", scaleInProtection)
```

Comment:
Is it worth moving the sub-logger creation on L117 to here, to avoid setting up two sub-loggers?

Reply:
Ah, good call. I didn't notice it wasn't being used in between.